swig/python/osgeo/gdal.py

```diff
@@ -1087,24 +1153,94 @@ def VectorTranslateOptions(options=None, format=None,
                     val += ','
                 val += item
             new_options += ['-select', val]
+        
+        if srcDatasetOpenOptions is not None:
```

Comment:
that can only work with the ogr2ogr binary, not gdal.VectorTranslate(), unless you do special processing of them when opening the dataset